Added Sysdig addon to EKS-A-Common

manuelbcd opened the pull request:
*Description of changes:*

Added Sysdig addon to the EKS-A-Common partners folder

elamaran11 commented on Validated_Partners/BareMetal_Partners.txt:
```diff
@@ -2,3 +2,4 @@
  kubecost
  suse
  hashicorp
+ sysdig
```
Please remove this. We update this with automation once your changes are approved, tested and merged

elamaran11 commented on Validated_Partners/nutanix_Partners.txt:
```diff
@@ -2,3 +2,4 @@
  kubecost
  suse
  hashicorp
+ sysdig
```
Please remove this. We update this with automation once your changes are approved, tested and merged

elamaran11 commented on Validated_Partners/rover_Partners.txt:
```diff
@@ -2,3 +2,4 @@
  kubecost
  suse
  hashicorp
+ sysdig
```
Please remove this. We update this with automation once your changes are approved, tested and merged

elamaran11 commented on Validated_Partners/vSphere_Partners.txt:
```diff
@@ -3,3 +3,4 @@
  suse
  newrelic
  hashicorp
+ sysdig
```
Please remove this. We update this with automation once your changes are approved, tested and merged

elamaran11 reviewed (commented):
@manuelbcd Thankyou for submitting the PR. Feedback on the PR 
1. Validated Partners in each deployment model is something our automation framework updates based SYSDIG's readiness to all platforms. Please remove your manual updates.
2. Hope the SYSDIG access yu shared to us will help us deploy this on all platforms and hope access is not limited.
3. We also need a testjob to validate SYSDIG like other partners have submitted. We arent approving PRs without test jobs to validate the partner software. Please submit a test job asap so we can start validation.

elamaran11 commented:
@manuelbcd We need a test job as we discussed to validate and pass this PR.

manuelbcd commented:
I'm sorry I did not commit all the files. Please review again @elamaran11

elamaran11 commented:
@manuelbcd The job fails with below error. The namespace specified should be `sysdig` for the test job.

```bash
flux-system   testers                  12d   False   Job/sysdig-agent/sysdig-agent-tester namespace not specified, error: namespaces "sysdig-agent" not found...
```

manuelbcd commented:
namespace switched to "sysdig" to match with the original name, thanks @elamaran11

elamaran11 commented:
@manuelbcd Job has an error with syntax. Appreciate if you can test it before pushing changes for test:
```bash
error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":""}}
```

manuelbcd commented:
- Added a more advanced functional test checking availability, readiness and audit port from the deployed agents
- Fixed the issue with the namespace name

Please check again @elamaran11

elamaran11 commented:
@manuelbcd The job fails again :


```bash
error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":""}}
```

elamaran11 commented:
@manuelbcd I can now see the test job working fine with your new changes :
```
nutanix@devstation:~$ k logs sysdig-agent-tester-hws8j -n sysdig
1. Checking readiness probe sysdig_pod=...
Waiting
2. Checking audit port sysdig_audit=...
Connecting
Success
```

elamaran11 commented:
@manuelbcd As discussed over slack. I hve validated the latest change you had an hour ago in all deployments models and everything works fine now. I can see the test job executing fine. Only change is you need to remove `sysdig` from Snow addons folder here https://github.com/aws-samples/eks-anywhere-addons/tree/main/eks-anywhere-snow/Addons/Partner/Sysdig because we have it in core, it syncs to all deployment models of EKS. Please do it asap to pass.

@shapirov103 I have validated the test job in all deployment models so please review and approve.

elamaran11 reviewed (commented):
Everything looks good. Only remove SYSDIG from Snow addons partner repo and Please push.

shapirov103 reviewed (commented):
Why do we still have https://github.com/aws-samples/eks-anywhere-addons/tree/main/eks-anywhere-snow/Addons/Partner/Sysdig? 
This deployment will interfere with common. I suspect on snow it will deployed twice. Can we validate the common config on snow? 
I also would to see the snow specific config dropped.
